qiskit/circuit/qpy_serialization.py

```diff
@@ -1503,7 +1512,7 @@ def _write_circuit(file_obj, circuit):
     instruction_buffer.close()
 
 
-def load(file_obj):
+def load(file_obj, custom_gate_dict=None):
```

Comment:
In general there's no need to include `dict` in the name of a variable - `map` would be more appropriate because it describes the purpose not the implementation, but even that is not necessary.  It could just be called `custom_gates`, and use the documentation to describe what is expected.  Also, what about instructions?

Reply:
- Okay, I shall keep the naming conventions in mind.
- To be honest, my fix was based on the issue which was raised. It addresses parameters not recognized by the QPY `load` function but on inspection I found that `Instruction.validate_parameter` had no validation as the default. I did make a parameter for the custom instructions too but removed in lieu of this definition.